service-statistics-unsd/src/main/java/org/oskari/statistics/plugins/unsd/UnsdDataParser.java

```diff
@@ -105,22 +117,33 @@ public StatisticalIndicatorDataDimension getTimeperiodDimensionFromIndicatorData
         }
     }
 
-    public static void parseTimePeriod(Map<Integer, Integer> countOfAreaCodesForYear, JSONObject response) {
+    public static void parseTimePeriod(Map<Integer, Set<Integer>> geoAreaCodesForYears, JSONObject response) {
         try {
             JSONArray data = response.getJSONArray(DATA_KEY);
             for (int i = 0; i < data.length(); i++) {
                 JSONObject o = (JSONObject) data.get(i);
                 Integer year = o.getInt(TIME_PERIOD_START_KEY);
-                countOfAreaCodesForYear.merge(year, 1, Integer::sum);
+                Integer geoAreaCode = o.getInt(GEO_AREA_CODE_KEY);
+
+                if(dataObjectValueTypeIsSupported(o)) {
+                    geoAreaCodesForYears.merge(year, new HashSet<Integer>(Arrays.asList(geoAreaCode)), (oldSet,newSet) -> {
+                        return Stream.of(oldSet,newSet).flatMap(x -> x.stream()).collect(Collectors.toSet());
+                    });
+                }
             }
         } catch (JSONException e) {
             LOG.error("Error parsing time period selectors for indicator: " + e.getMessage(), e);
         }
     }
     
+    private static boolean dataObjectValueTypeIsSupported(JSONObject o) throws JSONException {
+        return SUPPORTED_VALUE_TYPE.equals(o.getString("valueType"));
```

Comment:
Does the JSONObject here always have a value for `"valueType"` key? If not then o.optString(key); would be better. Also, maybe store the property name `"valueType"` in a field the same way as `DATA_KEY`, `TIME_PERIOD_START_KEY`, `GEO_AREA_CODE_KEY`.

Reply:
According to swagger document (https://unstats.un.org/SDGAPI/swagger/#!/Indicator/V1SdgIndicatorDataGet) valueType is marked as optional (as almost every field) but according to testing it seems that it is always present. I changed implementation to use optString() method as suggested.